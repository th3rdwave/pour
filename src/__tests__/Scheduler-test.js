// @flow

jest
  .mock('fs-extra')
  .mock('../Log')
  .mock('../Hash', () => ({
    sha1sum: () => 'a',
  }));

describe('Scheduler', () => {
  it('runs tasks in the right order', async () => {
    // Setup worker mock.
    jest.resetModules();
    // $FlowFixMe
    const run = jest.fn().mockResolvedValue();
    jest.doMock('../Worker', () => ({
      run,
    }));
    const Scheduler = require('../Scheduler');

    const graph = {
      node: { name: 'root' },
      edges: [
        {
          node: { name: 'app' },
          edges: [{ node: { name: 'shared' }, edges: [] }],
        },
        {
          node: { name: 'api' },
          edges: [{ node: { name: 'shared' }, edges: [] }],
        },
        { node: { name: 'shared' }, edges: [] },
      ],
    };

    await Scheduler.start({ dependencies: graph, rootDir: '', metaDir: '' });

    expect(run.mock.calls).toMatchSnapshot();
  });
});
