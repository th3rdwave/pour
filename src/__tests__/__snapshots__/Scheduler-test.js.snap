// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scheduler runs tasks in the right order 1`] = `
Array [
  Array [
    Object {
      "project": Object {
        "name": "root",
      },
      "rootDir": "",
    },
  ],
  Array [
    Object {
      "project": Object {
        "name": "shared",
      },
      "rootDir": "",
    },
  ],
  Array [
    Object {
      "project": Object {
        "name": "api",
      },
      "rootDir": "",
    },
  ],
  Array [
    Object {
      "project": Object {
        "name": "app",
      },
      "rootDir": "",
    },
  ],
]
`;
